sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseMetricReceiver.java

```diff
@@ -0,0 +1,70 @@
+package com.azure.monitor.opentelemetry.exporter.implementation.quickpulse;
+
+import com.azure.monitor.opentelemetry.exporter.implementation.MetricDataMapper;
+import com.azure.monitor.opentelemetry.exporter.implementation.logging.OperationLogger;
+import com.azure.monitor.opentelemetry.exporter.implementation.models.TelemetryItem;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+
+import java.util.Collection;
+import java.util.function.Consumer;
+
+import static com.azure.monitor.opentelemetry.exporter.implementation.utils.AzureMonitorMsgId.EXPORTER_MAPPING_ERROR;
+
+public class QuickPulseMetricReceiver implements Runnable {
+
+    private static QuickPulseHeaderInfo quickPulseHeaderInfo;
+    private QuickPulseMetricReader quickPulseMetricReader;
+    private QuickPulseDataCollector collector;
+    private static final OperationLogger metricReceiverLogger
+        = new OperationLogger(QuickPulseMetricReceiver.class, "Exporting metric");
+    private final MetricDataMapper mapper;
+    private final Consumer<TelemetryItem> quickPulseConsumer;
+
+    public QuickPulseMetricReceiver(QuickPulseMetricReader quickPulseMetricReader, MetricDataMapper metricDataMapper,
+        QuickPulseDataCollector collector) {
+        this.quickPulseMetricReader = quickPulseMetricReader;
+        this.mapper = metricDataMapper;
+        this.collector = collector;
+        this.quickPulseConsumer = telemetryItem -> {
+            if (this.collector.isEnabled()) {
+                this.collector.addOtelMetric(telemetryItem);
+            }
+        };
+    }
+
+    public static synchronized QuickPulseHeaderInfo getQuickPulseHeaderInfo() {
+        return quickPulseHeaderInfo;
+    }
+
+    public static synchronized void setQuickPulseHeaderInfo(QuickPulseHeaderInfo info) {
+        quickPulseHeaderInfo = info;
+    }
+
+    @Override
+    public void run() {
+        while (true) {
+
+            Collection<MetricData> metrics = quickPulseMetricReader.collectAllMetrics();
```

Comment:
This is an expensive operation - may want to introduce 1s delay when going from ping to post so we're not reading the last 5 seconds of metrics.